### PR TITLE
feat: add icon support in toggle button

### PR DESCRIPTION
**Pull request by abhishekashyap**
affects: @medly-components/core, @medly-components/theme

ISSUES CLOSED: #579

# PR Checklist

## Description
(Replace This Text: Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.)

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes and API changes)
- [ ] Build changes
- [ ] CI changes
- [ ] Document content changes
- [ ] Performance improvement
- [ ] Add missing tests
- [ ] Others (please describe)

## How has this been tested?
(Replace This Text: Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration.)

[ ] Test A

[ ] Test B

## What is the current behaviour?
(Replace This Text: Please describe the current behaviour you are modifying, or link a relevant issue.)

## What is the new behaviour?
(Replace This Text: Please describe the expected new behaviour.)

## Does this PR introduce a breaking change?

- [ ] Yes
- [ ] No

**Note:** (Replace This Text: If this PR contains a breaking change please describe the impact and migration path for existing application.)

## Additional context
(Replace This Text: Please describe any other related information or add screenshots of the PR.)

## Checklist

- [x] My code follows the style guidelines of this project

- [x] I have performed a self-review of my own code

- [x] I have commented my code, particularly in hard-to-understand areas

- [x] I have made corresponding changes to the documentation

- [x] My changes generate no new warnings

- [x] I have added tests that prove my fix is effective or that my feature works

- [x] New and existing unit tests pass locally with my changes

- [ ] Any dependent changes have been merged and published in downstream modules

Screenshots: 

Screenshot 2022-06-21 at 12 49 27 PM

Screenshot 2022-06-21 at 12 49 19 PM

### CODE REVIEW

File: packages/core/src/components/Toggle/Toggle.tsx [hunk 1]
```diff
@@ -1,14 +1,26 @@
...
+            label,
+            required,
+            labelPosition,
+            labelVariant,
+            labelWeight,
+            labelColor,
+            fullWidth,
+            onChange,
+            iconLeft,
+            iconRight,
```
Comment (PulkitBanta) [inline packages/core/src/components/Toggle/Toggle.tsx:21]: Can we use `OnIcon` and `OffIcon` to be more clear? Also this is the naming convention in the styled file, we can keep it consistent here

File: packages/core/src/components/Toggle/types.ts [hunk 2]
```diff
@@ -26,4 +27,8 @@ export interface ToggleProps extends InputProps {
     labelWeight?: FontWeights;
     /** Use if required in any form for example user needs to check 'I Agree' */
     required?: boolean;
+    /** Icon to show when toggle is in OFF state */
+    iconRight?: ReactElement;
+    /** Icon to show when toggle is in ON state */
+    iconLeft?: ReactElement;
```
Comment (PulkitBanta) [inline packages/core/src/components/Toggle/types.ts:33]: Can we use OnIcon and OffIcon to be more clear? Also this is the naming convention in the styled file, we can keep it consistent here

Comment (PulkitBanta) [inline packages/core/src/components/Toggle/types.ts:33]: Can we use OnIcon and OffIcon to be more clear? Also this is the naming convention in the styled file, we can keep it consistent here

Comment (abhishekashyap) [reply packages/core/src/components/Toggle/types.ts:33]: The reasoning for this was, while `offIcon` would make sense, but `onIcon` will cause conflict with `onClick`, `onScroll` and all the other actions starting with `on-`

Comment (samueljohnprasad) [reply packages/core/src/components/Toggle/types.ts:33]: can please add the screenshots of the toggle in the PR description.
@abhishekashyap

Comment (abhishekashyap) [reply packages/core/src/components/Toggle/types.ts:33]: Yes, I have added the screenshots. You can also view the chromatic storybooks deployment here: https://610a8c2fe7a1bd0039a44070-cvysngcsgn.chromatic.com/?path=/story/core--toggle-with-icon

File: packages/core/src/components/Toggle/Toggle.styled.tsx [hunk 3]
```diff
@@ -24,27 +25,58 @@ export const Wrapper = styled(FieldWithLabel.Field)<ToggleWrapperProps>`
...
+`;
+
+export const OffIcon = styled('div')<ToggleWrapperProps>`
+    color: ${({ theme }) => theme.toggle.iconColor};
+    transition: 0.2s ease-in-out;
     position: absolute;
-    background-color: ${({ theme }) => theme.toggle.bgColor};
+    right: 0;
+    svg {
+        height: ${props => `calc(${getHeight(props)} - 10px) `};
```
Comment (gmukul01) [inline packages/core/src/components/Toggle/Toggle.styled.tsx:47]: Could you please rems

File: packages/core/src/components/Toggle/Toggle.styled.tsx [hunk 4]
```diff
@@ -24,27 +25,58 @@ export const Wrapper = styled(FieldWithLabel.Field)<ToggleWrapperProps>`
...
-        content: '';
-        margin: 3px;
-        display: block;
-        border-radius: 50%;
-        background: ${({ theme }) => theme.colors.white};
-        box-shadow: 1px 3px 3px 1px rgba(0, 0, 0, 0.2);
-        transition: 0.2s;
-        width: ${props => `calc(${getHeight(props)} - 6px) `};
-        height: ${props => `calc(${getHeight(props)} - 6px) `};
+export const OnIcon = styled('div')<ToggleWrapperProps>`
```
Comment (gmukul01) [inline packages/core/src/components/Toggle/Toggle.styled.tsx:54]: You can move few of the common style from OffIcon and OnIcon

**Review by gmukul01 — CHANGES_REQUESTED** [review]

File: packages/core/src/components/Toggle/Toggle.styled.tsx (hunk 4 above)
Comment (abhishekashyap) [reply packages/core/src/components/Toggle/Toggle.styled.tsx:59]: Done, thanks!

File: packages/core/src/components/Toggle/Toggle.styled.tsx (hunk 3 above)
Comment (abhishekashyap) [reply packages/core/src/components/Toggle/Toggle.styled.tsx]: Yes, made the change.

**Review by gmukul01 — APPROVED** [review]

**Pull request merged**